cli/flags.go

```diff
@@ -433,6 +481,14 @@ func init() {
 			visitFlags(cmd, base.AddToFlagMap)
 		})
 
+		// If any of the security flags have been set, clear the insecure
+		// setting. Note that we do the inverse when the --insecure flag is
+		// set. See insecureValue.Set().
+		if cliContext.SSLCA != "" || cliContext.SSLCAKey != "" ||
+			cliContext.SSLCert != "" || cliContext.SSLCertKey != "" {
+			cliContext.Insecure = false
+		}
+
 		cliContext.Addr = net.JoinHostPort(connHost, connPort)
```

Comment:
Listening on `:<port>` is not the same thing as `localhost:<port>`. The former listens on all available interfaces. See https://golang.org/pkg/net/#Listen.